src/entities/trades/gnosis-protocol/CoWTrade.ts

```diff
@@ -289,9 +257,12 @@ export class CoWTrade extends Trade {
         ? CurrencyAmount.nativeCurrency(quoteResponse.quote.buyAmount.toString(), chainId)
         : new TokenAmount(tokenOut, quoteResponse.quote.buyAmount.toString())
 
+      // CoW Swap doesn't charge any fee
+      const fee = new Percent(ZERO, ONE)
+
       const feeAmount = Currency.isNative(currencyIn)
-        ? CurrencyAmount.nativeCurrency(quoteResponse.quote.feeAmount.toString(), chainId)
-        : new TokenAmount(currencyIn as Token, quoteResponse.quote.feeAmount.toString())
+        ? CurrencyAmount.nativeCurrency('0', chainId)
+        : new TokenAmount(currencyIn as Token, '0')
```

Comment:
MINOR, non-blocking
We could abstract this check into a function as we're duplicating this exact same code at line 191